gpt4all-bindings/rust/src/wrappers/embedding/embedding_model.rs

```diff
@@ -0,0 +1,143 @@
+use std::ffi::{c_char, CString};
+use std::ptr::{null, null_mut};
+use std::slice;
+
+use crate::bindings::{llmodel_embed, llmodel_free_embedding, llmodel_model, llmodel_model_destroy};
+use crate::wrappers::embedding::domain::{Embedding, EmbeddingOptions};
+use crate::wrappers::embedding::errors::EmbeddingError;
+use crate::wrappers::ffi_utils::string_from_ptr;
+
+
+/// Wrapper for handling embedding operations.
+pub struct EmbeddingModel {
+    /// The raw model pointer.
+    raw_model: llmodel_model
+}
+
+
+impl EmbeddingModel {
+    /// Creates a new `EmbeddingModel` instance.
+    pub fn new(raw_model: llmodel_model) -> Self {
+        Self {  raw_model }
+    }
+
+    /// Creates an embedding for the given options.
+    ///
+    /// # Arguments
+    ///
+    /// * `embedding_options` - Options for creating the embedding.
+    ///
+    /// # Returns
+    ///
+    /// * `Ok(Embedding)` if the embedding creation is successful.
+    /// * `Err(EmbeddingError)` if an error occurs during the process.
+    pub fn create_embedding(&self, embedding_options: EmbeddingOptions) -> Result<Embedding, EmbeddingError> {
+        // Collect texts into CString objects
+        let texts: Vec<CString> = embedding_options.texts
+            .iter()
+            .map(|text| CString::new(text.as_str()).expect("Failed to convert text to CString"))
```

Comment:
Would it be better to return an `EmbeddingError` variant if the input text is not a valid `CString` (rather than panicking)?